frame/examples/offchain-worker-ping-pong/src/lib.rs

```diff
@@ -0,0 +1,638 @@
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Offchain Worker entry point.
+		///
+		/// By implementing `fn offchain_worker` you declare a new offchain worker.
+		/// This function will be called when the node is fully synced and a new best block is
+		/// successfully imported.
+		/// Note that it's not guaranteed for offchain workers to run on EVERY block, there might
+		/// be cases where some blocks are skipped, or for some the worker runs twice (re-orgs),
+		/// so the code should be able to handle that.
+		/// You can use `Local Storage` API to coordinate runs of the worker.
+		fn offchain_worker(block_number: T::BlockNumber) {
+			// Note that having logs compiled to WASM may cause the size of the blob to increase
+			// significantly. You can use `RuntimeDebug` custom derive to hide details of the types
+			// in WASM. The `sp-api` crate also provides a feature `disable-logging` to disable
+			// all logging and thus, remove any logging from the WASM.
+			log::info!("Hello World from offchain workers!");
+
+			// Since off-chain workers are just part of the runtime code, they have direct access
+			// to the storage and other included pallets.
+			//
+			// We can easily import `frame_system` and retrieve a block hash of the parent block.
+			let parent_hash = <system::Pallet<T>>::block_hash(block_number - 1u32.into());
+			log::debug!("Current block: {:?} (parent hash: {:?})", block_number, parent_hash);
+
+			// if `NextUnsignedAt` allows, try to send some unsigned pong
+			let next_unsigned_at = <NextUnsignedAt<T>>::get();
+			if next_unsigned_at <= block_number {
+				// we choose which kind of unsigned pong based on block_number
+				let unsigned_type = block_number % 3u32.into();
+				if unsigned_type == Zero::zero() {
+					if let Err(e) = Self::ocw_pong_raw_unsigned(block_number) {
+						log::error!("Error: {}", e);
+					}
+				} else if unsigned_type == T::BlockNumber::from(1u32) {
+					// node needs to be loaded with keys as the payload will be signed
+					if let Err(e) = Self::ocw_pong_unsigned_for_any_account(block_number) {
+						log::error!("Error: {}", e);
+					}
+				} else if unsigned_type == T::BlockNumber::from(2u32) {
+					// node needs to be loaded with keys as the payload will be signed
+					if let Err(e) = Self::ocw_pong_unsigned_for_all_accounts(block_number) {
+						log::error!("Error: {}", e);
+					}
+				}
+			}
+
+			// try to send a pong_signed (node needs to be loaded with keys, account needs to be funded)
+			if let Err(e) = Self::ocw_pong_signed() {
+				log::error!("Error: {}", e);
+			}
+		}
+
+		/// clean Pings
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			Pings::<T>::kill();
+			Weight::zero()
+		}
+	}
+
+	/// A public part of the pallet.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
```

Comment:
```suggestion
		#[pallet::weight(0)]
```

Or just remove them by using dev-mode.

Reply:
Added `dev_mode`.